library/src/updater.rs

```diff
@@ -130,22 +131,29 @@ pub fn init(
 
     let libapp_path = libapp_path_from_settings(&app_config.original_libapp_paths)?;
     debug!("libapp_path: {:?}", libapp_path);
-    set_config(
+    let set_config_result = set_config(
         app_config,
         file_provider,
         libapp_path,
         &config,
         NetworkHooks::default(),
     );
 
-    let _ = with_config(|config| {
-        UpdaterState::load_or_new_on_error(
-            &config.storage_dir,
-            &config.release_version,
-            config.patch_public_key.as_deref(),
-        )
-        .on_init()
-    });
+    match set_config_result {
+        Err(err) => {
+            warn!("{}", err);
+        }
+        Ok(()) => {
+            let _ = with_config(|config| {
+                UpdaterState::load_or_new_on_error(
+                    &config.storage_dir,
+                    &config.release_version,
+                    config.patch_public_key.as_deref(),
+                )
+                .on_init()
+            });
+        }
```

Comment:
I would early-return in the Err case, so this block doesn't need to be nested.

Reply:
Done